### PR TITLE
Secrets 🔐 PR 6 - Update Oracle properties

### LINKED ISSUE #18035: Support Oracle driver SSL key and trust stores as secrets
#13552 provided a stopgap support for SSL in our Oracle driver, but it required setting JVM parameters for the key and trust stores.  This is suboptimal for a number of reasons.  It would be best to support specifying the keystore and/or truststore as Metabase secrets.

### PR DESCRIPTION
**Pull request by jeff303**
Adding keystore and truststore secret properties to Oracle YAML configuration

For more context see epic: #18036

### CODE REVIEW

File: src/metabase/models/secret.clj [hunk 1]
```diff
@@ -32,29 +33,38 @@
...
+  that temp file will be returned."
   {:added "0.42.0"}
   [{:keys [id ^bytes value] :as secret}]
   (if (= :file-path (:source secret))
     (let [secret-val          (value->string secret)
           ^File existing-file (File. secret-val)]
       (if (.exists existing-file)
         existing-file
-        (throw (ex-info (tru "Secret {0} points to non-existent file: {1}" id secret-val)
+        (throw (ex-info (tru "Secret {0} points to non-existent file: {1}" (if id id "") secret-val)
```
Comment (dpsutton) [inline src/metabase/models/secret.clj:59]: `(or id "")`

File: src/metabase/models/secret.clj [hunk 2]
```diff
@@ -32,29 +33,38 @@
...
                  {:secret-id id
                   :file-path secret-val}))))
     (let [^File tmp-file (doto (File/createTempFile "metabase-secret_" nil)
                            ;; make the file only readable by owner
                            (.setReadable false false)
                            (.setReadable true true)
                            (.deleteOnExit))]
-      (log/tracef "Creating temp file for secret %d value at %s" id (.getAbsolutePath tmp-file))
+      (log/tracef "Creating temp file for secret %s value at %s" (if id id "") (.getAbsolutePath tmp-file))
```
Comment (dpsutton) [inline src/metabase/models/secret.clj:67]: `(or id "")`

File: test/metabase/test/data/interface.clj [hunk 3]
```diff
@@ -190,7 +198,14 @@
...
   ;; take up to last 30 characters because databases like Oracle have limits on the lengths of identifiers
-  (apply str (take-last 30 (str/replace (str/lower-case (str database-name \_ table-name)) #"-" "_"))))
+  (-> (or *database-name-override* database-name)
+      (str \_ table-name)
+      str/lower-case
+      (str/replace #"-" "_")
+      (->>
+        (take-last 30)
+        (apply str)))
+  #_(apply str (take-last 30 (str/replace (str/lower-case (str (or *database-name-override* database-name) \_ table-name)) #"-" "_"))))
```
Comment (dpsutton) [inline test/metabase/test/data/interface.clj:208]: should we get rid of the commented out code?

File: modules/drivers/oracle/test/metabase/test/data/oracle.clj [hunk 4]
```diff
@@ -77,6 +89,38 @@
           session-schema
           (tx/db-qualified-table-name database-name table-name)))
 
+(defonce ^:private oracle-test-dbs-created-by-this-instance (atom #{}))
+
+(defn- destroy-test-database-if-created-in-different-session
+  "For Oracle, we have a randomly selected `session-schema`, in order to allow different test runs to proceed
+  independently. If we happen to have a test-database already in our app DB, that was created under a different session,
+  then we should just destroy it, since we won't be able to resync it anyway."
```
Comment (dpsutton) [inline modules/drivers/oracle/test/metabase/test/data/oracle.clj:97]: is it not better to have the creator clean up after themselves? I'm not following 100% how this issue arises.

Comment (jeff303) [reply modules/drivers/oracle/test/metabase/test/data/oracle.clj:97]: This line was written quite a while ago, now, but I had a fairly long Slack discussion with @camsaul  about it.  I'll find the context there and update the docstring accordingly.

File: modules/drivers/oracle/test/metabase/test/data/oracle.clj [hunk 5]
```diff
@@ -30,16 +34,24 @@
...
-
-(defmethod tx/dbdef->connection-details :oracle [& _] @connection-details)
+(defn- connection-details []
+  (let [details* {:host     (tx/db-test-env-var-or-throw :oracle :host)
+                  :port     (Integer/parseInt (tx/db-test-env-var-or-throw :oracle :port "1521"))
+                  :user     (tx/db-test-env-var-or-throw :oracle :user)
+                  :password (tx/db-test-env-var-or-throw :oracle :password)
+                  :sid      (tx/db-test-env-var-or-throw :oracle :sid)
+                  :ssl      (tx/db-test-env-var :oracle :ssl false)}]
+    (loop [details details*
```
[the remark this answers is not in the document]
Comment (jeff303) [reply modules/drivers/oracle/test/metabase/test/data/oracle.clj]: Done

**Review by dpsutton — APPROVED** [review]

**Pull request merged**